src/day06.rs

```diff
@@ -1,12 +1,126 @@
 // Potential improvements:
 //
 
-pub fn day06(_input_lines: &[Vec<String>], _solution_ver: &str) -> (String, String) {
-    let answer1 = 0;
-    let answer2 = 0;
+use std::collections::HashMap;
+use counter::Counter;
+
+use regex::Regex;
+
+pub fn day06(input_lines: &[Vec<String>], _solution_ver: &str) -> (String, String) {
+    let points = PointsList::new_from_puzzle_input(&input_lines[0]);
+    let answer1 = count_largest_area(&points);
+    let answer2 = count_included_distance(&points);
     (format!("{}", answer1), format!("{}", answer2))
 }
 
+#[derive(Debug, Clone, Copy, Hash, Eq, PartialEq)]
+struct Point {
+    x: i32,
+    y: i32,
+}
+
+impl Point {
+    fn new(x: i32, y: i32) -> Point {
+        Point{x, y}
+    }
+
+    fn manhattan_distance(self, other: &Point) -> i32 {
+        (self.x - other.x).abs() + (self.y - other.y).abs()
+    }
+}
+
+#[derive(Debug, Clone)]
+struct PointsList {
+    list: Vec<Point>
+}
+
+impl PointsList {
+    fn new_from_puzzle_input(input: &Vec<String>) -> PointsList {
+        let mut coords: Vec<Point> = vec!();
+        for coord in input {
+            let re = Regex::new(r"(\d+), (\d+)").unwrap();
+            let caps = re.captures(coord).unwrap();
+            let mut vals = caps
+            .iter()
+            .skip(1)
+            .map(|hit| hit.unwrap().as_str().parse::<i32>().unwrap());
+            let x = vals.next().unwrap();
+            let y = vals.next().unwrap();
+            coords.push(Point::new(x, y));
+        }
+        PointsList{list: coords}
+    }
+
+    fn x_min(&self) -> i32 {
+        self.list.clone().into_iter().min_by_key(|point| point.x).unwrap().x
```

Comment:
You're unnecessarily cloning the list here, you could also use map and min but it's not really better:
```suggestion
        self.list.iter().map(|&point| point.x).min().unwrap()
```